jsonapi/unmarshal.go

```diff
@@ -227,9 +224,13 @@ func setRelationshipIDs(relationships map[string]Relationship, target UnmarshalI
 }
 
 func checkType(incomingType string, target UnmarshalIdentifier) error {
-	actualType := getStructType(target)
-	if incomingType != actualType {
-		return fmt.Errorf("Type %s in JSON does not match target struct type %s", incomingType, actualType)
+	castedTarget, ok := target.(MarshalIdentifier)
```

Comment:
if you save method getStructType, you no need to cast here

Reply:
agree, restored getStructType and removed this casting